src/main/java/io/zeebe/containers/ZeebeTopologyWaitStrategy.java

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * Creates a new topology wait strategy for a single partition, which is the Zeebe default as
+   * well.
+   */
+  public ZeebeTopologyWaitStrategy() {
+    this(1);
+  }
+
+  /**
+   * Creates a new topology wait strategy for the given number of leaders.
+   *
+   * @param minimumLeadersCount the minimum number of partitions with leaders to wait for
+   */
+  public ZeebeTopologyWaitStrategy(final int minimumLeadersCount) {
+    this(minimumLeadersCount, ZeebePort.GATEWAY.getPort());
+  }
+
+  /**
+   * Creates a new topology wait strategy for the given number of leaders and a custom gateway port.
+   * This should only be used if you have properly overridden the gateway port everywhere, and is
+   * only for advanced usage.
+   *
+   * @param minimumLeadersCount the minimum number of partitions with leaders to wait for
+   * @param gatewayPort the custom gateway port
+   */
+  public ZeebeTopologyWaitStrategy(final int minimumLeadersCount, final int gatewayPort) {
+    this.gatewayPort = gatewayPort;
+    this.minimumLeadersCount = minimumLeadersCount;
+    this.clientBuilderProvider = () -> ZeebeClient.newClientBuilder().usePlaintext();
+  }
+
+  /**
+   * Sets a new number of minimum expected leaders. The topology is marked complete only when there
+   * are is at least one leader per partitions, so you should set this number to the number of
+   * partitions in your cluster.
+   *
+   * @param minimumLeadersCount the new minimum leader count
+   * @return this strategy, for chaining
+   */
+  public ZeebeTopologyWaitStrategy forMinimumLeadersCount(final int minimumLeadersCount) {
+    this.minimumLeadersCount = minimumLeadersCount;
+    return this;
+  }
+
+  /**
+   * Sets a custom gateway port, only useful if you have configured the gateway port to be a
+   * different one.
+   *
+   * @param gatewayPort the new gateway port
+   * @return this strategy, for chaining
+   */
+  public ZeebeTopologyWaitStrategy forGatewayPort(final int gatewayPort) {
+    this.gatewayPort = gatewayPort;
+    return this;
+  }
+
+  /**
+   * Sets a custom client builder provider; this provider will be given the broker contact point
+   * based on the given target. This is mostly useful if you wish to enable TLS/SSL on your gateway,
+   * at which point you can configure whatever you want.
+   *
+   * <p>Caveat: as the default provider applies {@link ZeebeClientBuilder#usePlaintext()}, if you
+   * still wish to use plaintext, make sure to call it as well in your custom provider.
+   *
+   * @param clientBuilderProvider the new client builder provider
+   * @return this strategy, for chaining
+   */
+  public ZeebeTopologyWaitStrategy forBuilder(
+      final Supplier<ZeebeClientBuilder> clientBuilderProvider) {
+    this.clientBuilderProvider = clientBuilderProvider;
+    return this;
+  }
+
+  @Override
+  protected void waitUntilReady() {
+    final TopologyHolder latestTopology = new TopologyHolder();
```

Comment:
imo using `AtomicReference` when what we want is a mutable, final reference can be misleading, since here I don't care at all about the atomicity. my concern would be that someone not familiar with this pattern would look at this and go "why the need for atomicity? did I miss something? :thinking: "

side note, here the performance aspect is not super important, but I guess in some cases it could be too.